### PR TITLE
Issues Fixed

### LINKED ISSUE #113: EditProfile bug with email address changes
If the user edits his email with a crap email, it might break the synchronization with this email on firebase

### LINKED ISSUE #135: EditProfile inputs to sanitize
Sanitize all inputs and check in database if email-address / username is already taken

### LINKED ISSUE #141: can't clear fields when editing profile
For example, if i want to edit my username, the first character of my current username can't be deleted

### LINKED ISSUE #142: MyEvents and History lists on another user's profile shouldn't show private events

### LINKED ISSUE #144: keyboard disappears every time a key is pressed when selecting a country in register screen
makes it really hard to input anything

### LINKED ISSUE #145: change the keyboard shown when entering a phone number in register screen
the current keyboard doesn't have the + symbol making it impossible to enter a valid phone number
Screenshot_2024-05-14-17-29-15-036_com github se gomeet

### LINKED ISSUE #152: change search bar padding from 5 to 10
`    SearchBar(
        shape = RoundedCornerShape(10.dp),
        query = query.value,
        onQueryChange = { query.value = it },
        active = false,
        modifier = Modifier.fillMaxWidth().padding(start = 5.dp, end = 5.dp),` //change to 10

### PR DESCRIPTION
**Pull request by chorfiyoussef**
This PR introduces some fixes to issues.

- Fixes #152 by modifying the `GoMeetSearchBar`
- Fixes #145 by showing a Phone Keyboard
- Fixes #144 by introducing a `DropDownMenu`
- Fixes #142 by using the `public` attribute of `Event`
- Fixes #141, Fixes #135 and Fixes #113 by sanitizing all inputs and removing the possibilityy for the User to change its email

### CODE REVIEW

File: app/src/main/java/com/github/se/gomeet/ui/authscreens/register/RegisterNameCountryPhone.kt [hunk 1]
```diff
@@ -102,56 +109,52 @@ fun RegisterNameCountryPhone(
...
-
-        if (!countryValid && !firstClick) {
-          Text(text = "Country is not valid", color = Color.Red)
+        if (!validLastName && !firstClick) {
+          Text(text = "Last Name is not valid", color = Color.Red)
         }
 
         Spacer(modifier = Modifier.size(screenHeight / 60))
-
+        /*Column {
```
Comment (DanThePol) [inline app/src/main/java/com/github/se/gomeet/ui/authscreens/register/RegisterNameCountryPhone.kt:117]: Need to remove all this commented out code

File: app/src/main/java/com/github/se/gomeet/ui/mainscreens/profile/EditProfile.kt [hunk 2]
```diff
@@ -79,22 +83,35 @@ fun EditProfile(
...
   var isLoaded by remember { mutableStateOf(false) }
   val showPopup = remember { mutableStateOf(false) }
 
+  var firstClick by remember { mutableStateOf(true) }
+
   LaunchedEffect(Unit) {
-    currentUser.value = userViewModel.getUser(com.google.firebase.Firebase.auth.currentUser!!.uid)
+    allUsers = userViewModel.getAllUsers()
+    currentUser.value = allUsers!!.first { it.uid == Firebase.auth.currentUser!!.uid }
+    allUsers!!.minus(currentUser.value!!)
```
Comment (DanThePol) [inline app/src/main/java/com/github/se/gomeet/ui/mainscreens/profile/EditProfile.kt:112]: This should be allUsers = allUsers!!.minus(currentUser.value!!), .minus() only returns a list without the specified element (it doesn't modify any lists)

File: app/src/main/java/com/github/se/gomeet/ui/mainscreens/profile/EditProfile.kt [hunk 3]
```diff
@@ -244,85 +279,60 @@ fun EditProfile(
...
+                    onValueChange = { newValue -> firstName.value = newValue },
                     label = { Text("First Name") },
                     singleLine = true,
                     modifier = Modifier.fillMaxWidth(),
                     colors = textFieldColors)
-
+                if (!firstNameValid && !firstClick) {
+                  Text(text = "First Name is not valid", color = Color.Red)
+                }
                 Spacer(modifier = Modifier.size(16.dp))
```
Comment (DanThePol) [inline app/src/main/java/com/github/se/gomeet/ui/mainscreens/profile/EditProfile.kt:290]: There are still some spacers that don't depend on the screen height

File: app/src/main/java/com/github/se/gomeet/ui/mainscreens/profile/EditProfile.kt [hunk 4]
```diff
@@ -244,85 +279,60 @@ fun EditProfile(
...
-
+                if (!lastNameValid && !firstClick) {
```
Comment (DanThePol) [inline app/src/main/java/com/github/se/gomeet/ui/mainscreens/profile/EditProfile.kt:291]: This should probably be under the last name field (for now, this text appears above the last name field, but all the other error messages appear under their respective fields)

**Review by DanThePol — CHANGES_REQUESTED** [review]
A few changes need to be made before we can merge, but otherwise this is great code that solves multiple open issues.

File: app/src/main/java/com/github/se/gomeet/ui/authscreens/register/RegisterNameCountryPhone.kt (hunk 1 above)
Comment (chorfiyoussef) [reply app/src/main/java/com/github/se/gomeet/ui/authscreens/register/RegisterNameCountryPhone.kt:117]: My mistake! This is fixed.

File: app/src/main/java/com/github/se/gomeet/ui/mainscreens/profile/EditProfile.kt (hunk 4 above)
Comment (chorfiyoussef) [reply app/src/main/java/com/github/se/gomeet/ui/mainscreens/profile/EditProfile.kt:291]: You're right, thanks for the feedback ! This has been fixed

File: app/src/main/java/com/github/se/gomeet/ui/mainscreens/profile/EditProfile.kt (hunk 2 above)
Comment (chorfiyoussef) [reply app/src/main/java/com/github/se/gomeet/ui/mainscreens/profile/EditProfile.kt:112]: Indeed... This is fixed now

**Review by DanThePol — APPROVED** [review]
All discussed changes have been fixed (see conversation). Good PR, these were some well-needed fixes.

**Pull request merged**